pkg/cmd/pipeline/start.go

```diff
@@ -352,9 +354,51 @@ func (opt *startOptions) getInput(pipeline *v1alpha1.Pipeline) error {
 		}
 	}
 
+	if opt.UseParamDefaults && !opt.Last && opt.UsePipelineRun == "" {
+		if err = opt.includeDefaultParams(pipeline); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+// Include the default in opts.Params
+func (opt *startOptions) includeDefaultParams(pipeline *v1alpha1.Pipeline) error {
+	//Creating a slice of string to contain the param.Name already included opt.Params through -p or --param option
+	var optParamNames = make([]string, 0)
+	for _, optParam := range opt.Params {
+		optParamNames = append(optParamNames, strings.Split(optParam, "=")[0])
+	}
+
+	for _, param := range pipeline.Spec.Params {
+
+		if contains(optParamNames, param.Name) {
+			continue
+		}
+		if param.Default != nil {
+			if param.Type == "string" {
+				opt.Params = append(opt.Params, param.Name+"="+param.Default.StringVal)
+				continue
+			} else {
+				opt.Params = append(opt.Params, param.Name+"="+strings.Join(param.Default.ArrayVal, ","))
+				continue
+			}
+		}
+	}
+	sort.Slice(opt.Params, func(i, j int) bool { return opt.Params[i] < opt.Params[j] })
+	return nil
+}
+
+func contains(elements []string, str string) bool {
```

Comment:
This whole function would not be necessary with the map-based approach mentioned above.